builder/js/ncf.js

```diff
@@ -89,6 +89,9 @@ app.directive('constraint', function($http, $q, $timeout) {
     },
     require: 'ngModel',
     link: function(scope, elm, attrs, ctrl) {
+    if((scope.parameter.constraints.allow_empty_string)&&(scope.parameter.value==null)){
```

Comment:
should check for null but ( scope.parameter.value === undefined)

And ndever double equal ;)